### PR TITLE
Merge harvest-legacy, registry-pds3-catalog, and registry-mgr-legacy

### LINKED ISSUE #77: Merge all legacy registry components into one repo
### 💡 Description

These components are so tightly coupled and not receiving many upgrades, lets just merge them all together for easier automated integration testing, management, and builds.

### ⚔️ Parent Epic / Related Tickets

_No response_

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary
Merge all repos together and refactor our search-core into separate library

## ⚙️ Test Data and/or Report
```
[SUCCESS] Harvest Successful
[INFO] Check solr docs match expected
[SUCCESS] Solr Doc Diff Successful
[INFO] Registry Manager Load
curl -X POST -H Content-Type: text/xml -d @/Users/jpadams/registry-data//solr-docs/solr-docs/solr_doc_0.xml http://localhost:8983/solr/data/update/xslt?commit=true&tr=add-hierarchy.xsl
<?xml version="1.0" encoding="UTF-8"?>
<response>

<lst name="responseHeader">
  <int name="rf">1</int>
  <int name="status">0</int>
  <int name="QTime">391</int>
</lst>
</response>
[SUCCESS] Registry Manager Load Successful
```

## ♻️ Related Issues

Resolves #77

### CODE REVIEW

**Review by nutjob4life — CHANGES_REQUESTED** [review]
- The conflict markers left in the codeql YAML should definitely be addressed
- There's _lots_ of commented-out code that should just be deleted completely (or have additional comments saying why), but I'll leave that as an option to consider

This pull request is so big it actually made Chrome unusable! I could count multiple seconds for keystrokes to appear in review fields

Safari on macOS handled it better than Chrome, but it too was lethargic. However Firefox was the clear winner! It had no issues. I think I just found my new preferred "work browser"

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -52,7 +52,11 @@ jobs:
 #       mvn clean compile site -DskipTests
 
     - name: Perform CodeQL Analysis
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply .github/workflows/codeql-analysis.yml]: Fixed per https://github.com/NASA-PDS/registry-harvest-legacy/pull/80/commits/1bc79add5c802ac1c8f463b84a41b7ae4e2048af

File: catalog-legacy/src/main/java/gov/nasa/pds/citool/comparator/CatalogComparator.java [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+                            targetAttribute.getLineNumber());
+                    Object[] arguments = {"Element",
+                            targetAttribute.getIdentifier().getId()};
+                    DiffException de = new DiffException(source.getSourceURI(),
+                            source.getLineNumber(),
+                            targetAttribute.getLineNumber(),
+                            "compare.nonMatchingValues", diffRecords,
+                            arguments);
+                    problems.add(de);
+/*
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply catalog-legacy/src/main/java/gov/nasa/pds/citool/comparator/CatalogComparator.java]: resolved per https://github.com/NASA-PDS/registry-harvest-legacy/pull/80/commits/438ebcb33d20ceb8fc3cc5f3fc29912d58a13748

File: catalog-legacy/src/main/java/gov/nasa/pds/citool/CIToolComparator.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+                        lp.getArguments()));
+            }
+            else {
+                report.recordSkip(uri, lp);
+            }
+        } catch (Exception e) {
+            report.recordSkip(uri, e);
+        }
+        return label;
+/*
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply catalog-legacy/src/main/java/gov/nasa/pds/citool/CIToolComparator.java]: resolved per https://github.com/NASA-PDS/registry-harvest-legacy/pull/80/commits/438ebcb33d20ceb8fc3cc5f3fc29912d58a13748

File: catalog-legacy/src/main/java/gov/nasa/pds/citool/comparator/CatalogComparator.java [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+            }
+        }
+        //Report any attributes found in the target, but not in the source.
+        for (PointerStatement t : targetPointers) {
+            Object[] arguments = {"Pointer", t.getIdentifier().getId()};
+            problems.add(new LabelParserException(t.getSourceURI(),
+                    t.getLineNumber(), null,
+                    "compare.source.missingStatement", ProblemType.MISSING_ID,
+                    arguments));
+/*
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply catalog-legacy/src/main/java/gov/nasa/pds/citool/comparator/CatalogComparator.java]: resolved per https://github.com/NASA-PDS/registry-harvest-legacy/pull/80/commits/438ebcb33d20ceb8fc3cc5f3fc29912d58a13748

File: catalog-legacy/src/main/java/gov/nasa/pds/citool/comparator/CatalogComparator.java [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+            if (targetPointer == null) {
+                //Pointer was found in the source, but not in the target.
+                Object[] arguments = {"Pointer",
+                        source.getIdentifier().getId()};
+                problems.add(new LabelParserException(source.getSourceURI(),
+                        source.getLineNumber(), null,
+                        "compare.target.missingStatement",
+                        ProblemType.MISSING_ID,
+                        arguments));
+/*
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply catalog-legacy/src/main/java/gov/nasa/pds/citool/comparator/CatalogComparator.java]: resolved per https://github.com/NASA-PDS/registry-harvest-legacy/pull/80/commits/438ebcb33d20ceb8fc3cc5f3fc29912d58a13748

File: catalog-legacy/src/main/java/gov/nasa/pds/citool/comparator/CatalogComparator.java [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+            }
+        }
+        //Report any attributes found in the target, but not in the source.
+        for (AttributeStatement t : targetAttributes) {
+            Object[] arguments = {"Element", t.getIdentifier().getId()};
+            problems.add(new LabelParserException(t.getSourceURI(),
+                    t.getLineNumber(), null,
+                    "compare.source.missingStatement", ProblemType.MISSING_ID,
+                    arguments));
+/*
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply catalog-legacy/src/main/java/gov/nasa/pds/citool/comparator/CatalogComparator.java]: resolved per https://github.com/NASA-PDS/registry-harvest-legacy/pull/80/commits/438ebcb33d20ceb8fc3cc5f3fc29912d58a13748

File: catalog-legacy/src/main/java/gov/nasa/pds/citool/comparator/CatalogComparator.java [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+                            targetPointer.getLineNumber());
+                    Object[] arguments = {"Pointer",
+                            targetPointer.getIdentifier().getId()};
+                    DiffException de = new DiffException(source.getSourceURI(),
+                            source.getLineNumber(),
+                            targetPointer.getLineNumber(),
+                            "compare.nonMatchingValues", diffRecords,
+                            arguments);
+                    problems.add(de);
+/*
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply catalog-legacy/src/main/java/gov/nasa/pds/citool/comparator/CatalogComparator.java]: resolved per https://github.com/NASA-PDS/registry-harvest-legacy/pull/80/commits/438ebcb33d20ceb8fc3cc5f3fc29912d58a13748

File: catalog-legacy/src/main/java/gov/nasa/pds/citool/comparator/CatalogComparator.java [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+                    source.getIdentifier().getId());
+            if (targetAttribute == null) {
+                //Attribute was found in the source, but not in the target.
+                Object[] arguments = {"Element", source.getIdentifier().getId()};
+                problems.add(new LabelParserException(source.getSourceURI(),
+                        source.getLineNumber(), null,
+                        "compare.target.missingStatement",
+                        ProblemType.MISSING_ID,
+                        arguments));
+/*
```
[the remark this answers is not in the document]
Comment (jordanpadams) [reply catalog-legacy/src/main/java/gov/nasa/pds/citool/comparator/CatalogComparator.java]: resolved per https://github.com/NASA-PDS/registry-harvest-legacy/commit/438ebcb33d20ceb8fc3cc5f3fc29912d58a13748

**Comment by jordanpadams** [comment]
thanks for the PR review @nutjob4life !

**Comment by nutjob4life** [comment]
> thanks for the PR review @nutjob4life !

Happy to help @jordanpadams!

**Comment by jordanpadams** [comment]
Merging for now. Still need to do some more research to fix the branch testing.

**Pull request merged**